proposals/mmmm-spawn-let.md

```diff
@@ -0,0 +1,701 @@
+
+```swift
+spawn var x = nope() // error: 'spawn' can only be used with 'let' declarations
+```
+
+Other than having to be awaited to access its value, a `spawn let` behaves just like a typical `let`, as such it is not possible to pass it `inout` to other functions - simply because it is a `let`, and those may not be passed as `inout`.
+
+#### Declaring `spawn let` with patterns
+
+It is possible to create a `spawn let` where the left hand side is a pattern, e.g. a tuple, like this:
+
+```swift
+func left() async -> String { "l" }
+func right() async -> String { "r" }
+
+spawn let (l, r) = (left(), right())
+
+await l // at this point `r` is also assumed awaited-on
+```
+
+To understand the execution semantics of the above snippet, we can remember the sugaring rule that the right-hand side of a `spawn let` effectively is just a concurrently executing asynchronous closure:
+
+```swift
+spawn let (l, r) = {
+  return await (left(), right())
+  // -> 
+  // return (left await(), await right())
+}
+```
+
+meaning that the entire initializer of the `spawn let` is a single task, and if multiple asynchronous function calls are made inside it, they are performed one-by one. This is a specific application of the general rule of `spawn let` initializers being allowed to omit a single leading `await` keyword before their expressions. Because in this example, we invoke two asynchronous functions to form a tuple, the await can be moved outside of the expression, and that await is what is omitted in the short hand form of the `spawn let` that we've seen in the first snippet.
+
+This also means that as soon as we enter continue past the line of `await l` it is known that the `r` value also has completed sucessfully (and will not need to emit an "implicit await" which we'll discuss in detail below).
+
+Another implication of these semantics is that if _any_ piece of the initializer throws, any await on such pattern declared `spawn let` shall be considered throwing, as they are initialized "together". To visualize this, let us consider the following:
+
+```swift
+async let (yay, nay) = ("yay", throw Boom())
+try await yay // because the (yay, nay) initializer is throwing
+```
+
+Because we know that the right-hand side is simply a single closure, performing the entire initialization, we know that if any of the operations on the right hand size is throwing, the entire initializer will be considered throwing. As such, awaiting even the `yay` here must be ready for that initializer to have thrown and therefore must include the `try` keyword in addition to `await`.
+
+### Awaiting `spawn let` values
+
+Since `spawn let`s introduce constants that will be "filled in later" by their right-hand-side concurrently-executing task, refering to them must be covered by an `await` keyword:
```

Comment:
`refering` → `referring`

```suggestion
Since `spawn let`s introduce constants that will be "filled in later" by their right-hand-side concurrently-executing task, referring to them must be covered by an `await` keyword:
```